src/rviz/default_plugin/odometry_display.cpp

```diff
@@ -372,6 +382,26 @@ void OdometryDisplay::update(float /*wall_dt*/, float /*ros_dt*/)
 
   assert(arrows_.size() == axes_.size());
   assert(axes_.size() == covariance_property_->sizeVisual());
+
+  applyTransforms();
+}
+
+bool OdometryDisplay::applyTransforms()
+{
+  if(!ref_pose_)
+    return false;
+
+  Ogre::Vector3 position;
+  Ogre::Quaternion orientation;
+  if (!context_->getFrameManager()->transform(last_message_->header, *ref_pose_, position, orientation))
```

Comment:
Seems like `last_message_` is only used to store its header?
What if the involved frames in the odometry messages change over time?

Reply:
Yes its only used to store its header or mainly its timestamp. See above.

For me it is a very uncommon scenario to publish Odometry messages with different parent/child frame_id's to the same topic. Currently this wouldn't be handled properly.

Reply:
To handle this properly a scene node for each parent frame would be required, which would make everything more complex and definitively break API/ABI.

Reply:
I could add a warning when a Odometry message with a different parent frame_id arrives, which says that it should be published on another topic.